lib/src/context.dart

```diff
@@ -959,8 +959,12 @@ class Context {
   ///
   ///     path.fromUri('path/to/foo'); // -> 'path/to/foo'
   String fromUri(uri) {
-    if (uri is String) uri = Uri.parse(uri);
-    return style.pathFromUri(uri);
+    if (uri is String) {
+      return style.pathFromUri(Uri.parse(uri));
+    } else if (uri is Uri) {
+      return style.pathFromUri(uri);
+    }
```

Comment:
Rather than repeating this logic, could you add a `Uri _parseUri(String)` helper?